src/Annotations/ParameterAnnotations.php

```diff
@@ -25,7 +25,7 @@ public function __construct(private array $annotations)
      *
      * @param class-string<T> $className
      *
-     * @return array<int, T>
+     * @return T[]
```

Comment:
In phpdoc `T[]` is used for ordered lists. It is array where integer keys are incremented by 1 and key actually means order of value in array. If array size is 888, then last index should be 887.
In other case, we use `array<int, T>` to tell that array values is not ordered by its keys.
For instance, if we call `array_values()` on `array<int, T>`  below:
```
[
  0 =>'Im a first element',
  999 =>'Im second element',
  5 =>'Im a third element',
]
````
Result is an `T[]`. Values are not ordered by its key. Value with highest key 999 stays where we declared it.
```
[
  0 =>'Im a first element',
  1 =>'Im second element',
  2 =>'Im a third element',
]
```

In case of `ParameterAnnotations` collection, `$annotations` is actually a list. Even, class descriptor tells us that it is `A list of annotations that implement the ParameterAnnotation interface`

Additionally, I analysed usage of `ParameterAnnotations` and think that code do not respect keys of that array, only order matters. So, in my opinion we should use `T[]` in that case. Am I missing anything?

Reply:
`T[]` means `array<array-key, T>` (where array-key is int|string) - array with any keys (ordered or unordered)
`array<int, T>` - array with integer keys (ordered or unordered)

If it is ordered list (like a result of `array_values`), the proper type is `list<T>` - array with integer keys (ordered, from 0)

https://psalm.dev/docs/annotating_code/type_syntax/array_types/

Reply:
Thank you for explanation! 
Seems like passing numerable string as key in lists is allowed in both Psalm and PHPStan even with highest level of error reporting.
I sorry for such misunderstanding! 
Pushed commit with improved type safety for arrays!